github.go

```diff
@@ -69,30 +70,32 @@ func NewGithubClient(
 	connectionTimeout time.Duration,
 ) *GithubLocator {
 	var httpClient *http.Client
+	ctx, cancel := ensureTimeout(ctx, connectionTimeout)
+	defer cancel()
+
 	if token != "" {
 		src := oauth2.StaticTokenSource(
 			&oauth2.Token{AccessToken: token},
 		)
 		httpClient = oauth2.NewClient(ctx, src)
-		httpClient.Timeout = connectionTimeout
-	}
-
-	if httpClient == nil {
```

Comment:
if token is not set - we'll get `nil` client

Reply:
This is fine, the github client constructor checks if the httpClient is nil and uses the default one

Reply:
```go
	if httpClient == nil {
		httpClient = http.DefaultClient
	}
```

Reply:
default client comes w/out any timeout

Reply:
the timeout is controller by context, not on the http client